### DISCUSSION TITLE
Tutorial: How to get the same render in both Malt and Godot Engine

### DISCUSSION THREAD

**Discussion by panthavma** [bnpr/Malt#287] [Showcase]
Hi! I wrote an article on how to achieve the same result between Malt and Godot when using advanced rendering pipelines.

This uses Malt's custom pipelines and [OddLib](https://github.com/panthavma/oddlib) for easier setup in Godot

Link to the article: [https://panthavma.com/articles/godot-malt-pipeline-intro/](https://panthavma.com/articles/godot-malt-pipeline-intro/)
Link to the repo: [https://github.com/panthavma/20220424-godot-malt-pipeline](https://github.com/panthavma/20220424-godot-malt-pipeline)

![image](https://user-images.githubusercontent.com/22663538/164980808-041d6440-432c-4e37-92f9-183f58171ddd.png)

**Comment by pragma37** [comment]
Awesome!

Before Malt, I developed a (never released) addon for a very similar use case.
Being able to edit your assets while previewing something close to the final look can be a game changer.

Thank you for taking the time to write the tutorial. :)

I haven't looked too much into Godot, but I think it should be possible, indeed, to use some preprocessor magic to write Godot-style shaders in Malt.

**Reply by panthavma** [reply]
Thank you for making Malt! It's both super fun to use and REALLY useful once you get over the initial hurdle (which will get easier as time goes on and tutorials start appearing). I really feel like I can use my 3D knowledge interactively now!

Yeah, the godot shader language is similar to the rest, but in an effort to make it simpler to write shaders it's not compatible with the usual vertex/fragment workflow. I think a good amount of #DEFINE can certainly do the job, and have the malt pipeline straight up export a variation on the code when compiling shaders.
